### PR TITLE
Service outline

### PR DESCRIPTION
**Pull request by rish691**
This PR provides the basic structure of components for building alerting solution as requested part of feature request - https://github.com/hypertrace/hypertrace/issues/251

### CODE REVIEW

File: alerting-rule-manager/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+plugins {
```
Comment (kotharironak) [inline alerting-rule-manager/build.gradle.kts:1]: Pl. add the `Licence.txt` file at repo level.

File: settings.gradle.kts [hunk 2]
```diff
@@ -10,5 +10,6 @@ plugins {
     id("org.hypertrace.version-settings") version "0.2.0"
 }
 
-rootProject.name = "alerting"
-
+include(":metric-anomaly-detector")
+include(":alerting-rule-manager")
```
Comment (kotharironak) [inline settings.gradle.kts:14]: Should we rename this project as `metric-anomaly-task-producer`?

File: settings.gradle.kts [hunk 3]
```diff
@@ -10,5 +10,6 @@ plugins {
     id("org.hypertrace.version-settings") version "0.2.0"
 }
 
-rootProject.name = "alerting"
```
Comment (kotharironak) [inline settings.gradle.kts:13]: Will we have `hypertrace-alert-manager` /  `hypertrace-alerting-system` or `hypertrace-alert-evalutor` as root project name eventually?

Do you think, we should have the same name for our repo too like hypertrace-service / hypertrace-ingester ?

File: metric-anomaly-detector-api/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+plugins {
+    `java-library`
+    // id("org.hypertrace.avro-plugin")
```
Comment (kotharironak) [inline metric-anomaly-detector-api/build.gradle.kts:3]:
```suggestion
```

File: metric-anomaly-detector-api/src/main/avro/MetricAnomalyDetectionTask.avdl [hunk 5]
```diff
@@ -0,0 +1,4 @@
+@namespace("org.hypertrace.alerting.metric.anomaly.detection")
```
Comment (kotharironak) [inline metric-anomaly-detector-api/src/main/avro/MetricAnomalyDetectionTask.avdl:1]:
```suggestion
@namespace("org.hypertrace.metric.anomaly.detection.api")
```

File: metric-anomaly-detector-api/src/main/avro/MetricAnomalyDetectionTask.avdl [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+protocol MetricAnomalyDetectionTaskProtocol {
+  bytes rule;
```
Comment (kotharironak) [inline metric-anomaly-detector-api/src/main/avro/MetricAnomalyDetectionTask.avdl:3]:
```suggestion
 record MetricAnomalyDetectionTask {
    bytes rule;
  }
```

File: metric-anomaly-detector-api/build.gradle.kts (hunk 4 above)
Comment (kotharironak) [reply metric-anomaly-detector-api/build.gradle.kts:3]: The above plugin is required - id("org.hypertrace.avro-plugin"), right?

File: metric-anomaly-detector/src/main/resources/configs/common/application.conf [hunk 7]
```diff
@@ -0,0 +1,7 @@
+service.name = alerting-rule-manager
```
Comment (kotharironak) [inline metric-anomaly-detector/src/main/resources/configs/common/application.conf:1]: metric-anomaly-detector

File: metric-anomaly-detector/build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+dependencies {
+    implementation("org.hypertrace.core.serviceframework:platform-service-framework:0.1.23")
+    implementation("org.hypertrace.core.serviceframework:platform-metrics:0.1.23")
+
+    // Required for the GRPC clients.
+    runtimeOnly("io.grpc:grpc-netty:1.36.1")
+
+    implementation("com.typesafe:config:1.4.1")
+    implementation("io.confluent:kafka-avro-serializer:6.0.1")
```
Comment (kotharironak) [inline metric-anomaly-detector/build.gradle.kts:20]: Pl. keep only used one. Whenever you start using it, along with it, lets add deps.

File: build.gradle.kts [hunk 9]
```diff
@@ -1,27 +1,24 @@
 plugins {
     id("org.hypertrace.repository-plugin") version "0.4.0"
     id("org.hypertrace.ci-utils-plugin") version "0.3.0"
+    id("org.hypertrace.docker-java-application-plugin") version "0.9.0" apply false
+    id("org.hypertrace.docker-publish-plugin") version "0.9.0" apply false
```
Comment (kotharironak) [inline build.gradle.kts:5]: Can we remove the publish plugins for now? Once we have some working version, we will add it. Remove both - docker, and lib.

**Review by kotharironak — CHANGES_REQUESTED** [review]

File: metric-anomaly-detector-api/build.gradle.kts (hunk 4 above)
Comment (rish691) [reply metric-anomaly-detector-api/build.gradle.kts:3]: So this plugin performs a backward compatibility check with the last published version, so will enable it after first version artifact is published

File: metric-anomaly-task-manager/src/main/resources/configs/common/application.conf [hunk 10]
```diff
@@ -0,0 +1,7 @@
+service.name = metric-anomaly-rule-manager
```
Comment (kotharironak) [inline metric-anomaly-task-manager/src/main/resources/configs/common/application.conf:1]:
```suggestion
service.name = metric-anomaly-task-manager
```

File: metric-anomaly-task-manager/src/main/resources/configs/common/application.conf [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+service.admin.port = 8099
+
+main.class = org.hypertrace.alerting.metric.alerting.rule.manager.AlertingRuleManager
```
Comment (kotharironak) [inline metric-anomaly-task-manager/src/main/resources/configs/common/application.conf:4]: Let's change the repo name and package structure as follow-up PR.

**Review by kotharironak — DISMISSED** [review]
Approving this now. As discussed, let's change the repo name, and package struct for the task manager as follow up.

**Review by kotharironak — APPROVED** [review]

File: metric-anomaly-task-manager/src/main/resources/configs/common/application.conf (hunk 11 above)
Comment (rish691) [reply metric-anomaly-task-manager/src/main/resources/configs/common/application.conf:4]: yeah

**Pull request merged**